### PR TITLE
fix(kmodal): dismiss icon logic [beta]

### PR DESCRIPTION
**Pull request by kaiarrowood**
# Summary

## Vue 3

**Did you create a corresponding PR to add this change to the `beta` branch? (required)**

- [ ] **Yes**, here is a link to the PR: `[link to beta branch PR]`
- [ ] **No**, I did not create a PR for the `beta` branch and I have added the `port to beta branch` tag to this PR.
- [ ] **No**, because `[type your reasons]`

If you have questions, tag `@adamdehaven` or `@kaiarrowood`.

---

## PR Checklist

- [ ] Does not introduce dependencies
- [ ] **Functional:** all changes do not break existing APIs and if so, bump major version.
- [ ] **Tests pass:** check the output of yarn test packages/<Kongponent>
- [ ] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
- [ ] **Framework style:** abides by the essential rules in Vue's style guide
- [ ] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs), or leftover comments
- [ ] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

File: docs/components/modal.md [hunk 1]
```diff
@@ -130,9 +130,13 @@ Change the [appearance](/components/button.html#props) of the cancel button.
 
 Use this to hide the built-in cancel button (`false` by default).
 
-### showDismissIcon
+### hideDismissIcon
 
-Boolean for whether or not to display the 'X' in the upper right corner to dismiss the dialog (`false` by default).
+When using the `header-image` slot we display a dismiss 'X' button in the upper right corner of the dialog. Boolean for hiding this button (`false` by default).
```
Comment (kaiarrowood) [inline docs/components/modal.md:135]:
```suggestion
When using the `header-image` slot we display a dismiss 'X' button in the upper right corner of the dialog. Set this prop to `true` to hide the.
```

File: docs/components/modal.md [hunk 2]
```diff
@@ -130,9 +130,13 @@ Change the [appearance](/components/button.html#props) of the cancel button.
 
 Use this to hide the built-in cancel button (`false` by default).
 
-### showDismissIcon
+### hideDismissIcon
 
-Boolean for whether or not to display the 'X' in the upper right corner to dismiss the dialog (`false` by default).
+When using the `header-image` slot we display a dismiss 'X' button in the upper right corner of the dialog. Set this prop to `true` to hide the.
```
Comment (jillztom) [inline docs/components/modal.md:135]: this section is not complete. 

When using the `header-image` slot we display a dismiss 'X' button in the upper right corner of the dialog. Set this prop to `true` to hide the button (`false` by default).

**Review by jillztom — CHANGES_REQUESTED** [review]
Referenced some documentation correction.

File: docs/components/modal.md (hunk 2 above)
Comment (adamdehaven) [reply docs/components/modal.md:135]: @kaiarrowood these will need to go to `main` as well 😬

Comment (kaiarrowood) [inline docs/components/modal.md:135]:
```suggestion
When using the `header-image` slot we display a dismiss 'X' button in the upper right corner of the dialog. Set this prop to `true` to hide the button (`false` by default).
```

**Review by adamdehaven — APPROVED** [review]

**Pull request merged**